### DISCUSSION TITLE
warning on invalid dataattribute

### DISCUSSION THREAD

**Discussion by bkbartk** [frenck/spook#655] [Ideas] [answered]
One of my services was updated and the dataattribute name was changed
```
service: servicename
data:
  old_attribute: value
```
now, My automation crashes
![image](https://github.com/frenck/spook/assets/17070805/de4bf052-f660-409e-857d-2d985fbd4da7)
but spook does not detect.
Is is possible to detect this, So I know I need to fix it?

**Comment by frenck** [comment]
Without details on the service, I cannot tell you if we could detect this or not.

../Frenck

**Comment by bkbartk** [comment]
it is actually an esphome service
the old way of calling was
```
service: esphome.lamp_kamer_voor_play_rtttl
data:
  song_str: some value
```
and the new way is
```
service: esphome.lamp_kamer_voor_rtttl_play
data:
  tone: some value
```
so I got the repair for the service name change
so I changed this everywhere in `automations.yaml`
but I missed the fact that `song_str` was replaced by `tone` 
and therefor my automation stopped at the first action and did not continue(see screenshot)
not sure why `continue_on_error: true` didn't help, but that might be as designed.

**Comment by frenck** [comment, marked as answer]
These are ESPHome-based services; they are user-created, there is no info available for this.

../Frenck

**Comment by bkbartk** [comment]
thnx,
I was under the impression that spook would also validate(non-default) services at runtime.

**Reply by frenck** [reply]
It validates the called services exists.
